src/client/services/board/persistence/persistence.js

```diff
@@ -65,6 +65,50 @@ var PersistenceService = SparkBase.extend({
       }.bind(this));
   },
 
+
+  /**
+   * Adds a snapshot image of the board
+   *
+   * @param {Conversation} conversation - the current conversation that the board belongs
+   * @param {Board~Channel} channel
+   * @param {File} image
+   * @returns {Promise<Board~Channel>}
+   */
+  addSnapshotImage: function addSnapshotImage(conversation, channel, image) {
+    var imageScr;
+    return this.spark.board._uploadImage(conversation, image)
+      .then(function encryptScr(scr) {
+        imageScr = scr;
+        return this.spark.encryption.encryptScr(imageScr, conversation.defaultActivityEncryptionKeyUrl);
+      }.bind(this))
+      .then(function attachEncryptedScr(encryptedScr) {
+        imageScr.encryptedScr = encryptedScr;
+        return encryptedScr;
+      }.bind(this))
+      .then(function addSnapshotToChannel() {
+        var imageBody = {
+          image: {
+            url: imageScr.loc,
+            height: image.height || 900,
+            width: image.width || 1600,
+            mimeType: image.type || 'image/png',
+            scr: imageScr.encryptedScr,
+            encryptionKeyUrl: conversation.defaultActivityEncryptionKeyUrl,
+            fileSize: image.size
+          }
+        };
+        return this.spark.request({
+          method: 'PATCH',
```

Comment:
I'm doing a PATCH to update the image of the channel (this is specified by the server)
```
{
    "image": {}
}
```
The channel DTO is something like this
```JSON
{
    "channelId": "fd48a240-23eb-11e4-be51-56aa0a6fd5da",
    "channelUrl": "https://board-a.wbx2.com/board/api/v1/channels/fd48a240-23eb-11e4-be51-56aa0a6fd5da",
    "aclUrl": "https://acl-a.wbx2.com/acl/api/v1/acls",
    "parentId": null,
    "type": "board",
    "creatorId": "",
    "createdTime": 1435278171139,
    "properties": { "penColor": "black"},
    "defaultEncryptionKeyUrl" : "", 
    "kmsResourceUrl": "https://encryption-a.wbx2.com/encryption/api/v1/resources/",
    "contentLastUpdatedTime" : 1435278182139,   
    "hiddenSpaceUrl": "https://files-api-a.wbx2.com/v1/spaces/",
    "conversationUrl":null,
    "image":{
          "updatedTime": 1435278182139,
          "updateInterval": 360000,
          "url": "https://files-api-a.wbx2.com/v1/spaces/bytes",
          "height": 1405,
          "width": 1080,
          "fileSize": 4691565,
          "mimeType": "image/png",
          "scr": "",
          "encryptionKeyUrl" : ""
        },
    "aclUrlLink":"https://acl-a.wbx2.com/acl/api/v1/acls",
    "imageUpdateInterval":36000
}
```